### ISSUE TITLE
how to train train_siamRPN ? 

### ISSUE THREAD

**Issue by shiyongde** [researchmm/SiamDW#5]
how to train train_siamRPN ? 
just change  MODEL:"SiamRPNRes22"  in yaml ? 
I think  it need rpn loss function .

**Comment by JudasDie** [comment]
> 
> 
> how to train train_siamRPN ?
> just change MODEL:"SiamRPNRes22" in yaml ?
> I think it need rpn loss function .

1) Only SiamFC training code is provided now. 
2) If you want to train SiamRPN, dataloader, loss function, connect head, training strategy should be modified.
3) YTB data is too large to upload. But you can try other data like GOT10K.
3) SiamRPN is much more severly influenced by hyper-parameter. Please pay attention to this.

**Comment by lzx1413** [comment]
> > how to train train_siamRPN ?
> > just change MODEL:"SiamRPNRes22" in yaml ?
> > I think it need rpn loss function .
> 
> 1. Only SiamFC training code is provided now.
> 2. If you want to train SiamRPN, dataloader, loss function, connect head, training strategy should be modified.
> 3. YTB data is too large to upload. But you can try other data like GOT10K.
> 4. SiamRPN is much more severly influenced by hyper-parameter. Please pay attention to this.

Does The 4 mean that you are using the post process parameter to overfit the test dataset?

**Comment by JudasDie** [comment]
> > > how to train train_siamRPN ?
> > > just change MODEL:"SiamRPNRes22" in yaml ?
> > > I think it need rpn loss function .
> > 
> > 
> > 
> > 1. Only SiamFC training code is provided now.
> > 2. If you want to train SiamRPN, dataloader, loss function, connect head, training strategy should be modified.
> > 3. YTB data is too large to upload. But you can try other data like GOT10K.
> > 4. SiamRPN is much more severly influenced by hyper-parameter. Please pay attention to this.
> 
> Does The 4 mean that you are using the post process parameter to overfit the test dataset?

In fact,  almost all tracking algorithms are sensitive to hyper-parameters. Reasonable hyper-parameter is important to trackers.

**Comment by lzx1413** [comment]
> > > > how to train train_siamRPN ?
> > > > just change MODEL:"SiamRPNRes22" in yaml ?
> > > > I think it need rpn loss function .
> > > 
> > > 
> > > 
> > > 1. Only SiamFC training code is provided now.
> > > 2. If you want to train SiamRPN, dataloader, loss function, connect head, training strategy should be modified.
> > > 3. YTB data is too large to upload. But you can try other data like GOT10K.
> > > 4. SiamRPN is much more severly influenced by hyper-parameter. Please pay attention to this.
> > 
> > 
> > Does The 4 mean that you are using the post process parameter to overfit the test dataset?
> 
> In fact, almost all tracking algorithms are sensitive to hyper-parameters. Reasonable hyper-parameter is important to trackers.

and to each test video.

**Comment by ngunauj** [comment]
pay attention

**Comment by JudasDie** [comment]
> 
> 
> how to train train_siamRPN ?
> just change MODEL:"SiamRPNRes22" in yaml ?
> I think it need rpn loss function .

SiamRPN+ training code have been uploaded. Training-Testing-Tuning is included in siamese_tracking/onekey.py. There is a tutorial for how to implement your ideas in our framework. See [train.md.](https://github.com/researchmm/SiamDW/blob/master/lib/tutorials/train.md)
